src/libraries/Microsoft.Extensions.Logging/tests/Common/LoggerFactoryTest.cs

```diff
@@ -213,6 +216,171 @@ public void TestInvalidActivityTrackingOptions()
             );
         }
 
+        [Fact]
+        public void TestActivityTrackingOptions_ShouldAddBaggageItemsAsNewScope_WhenBaggageOptionIsSet()
+        {
+            var loggerProvider = new ExternalScopeLoggerProvider();
+
+            var loggerFactory = LoggerFactory.Create(builder =>
+            {
+                builder
+                    .Configure(o => o.ActivityTrackingOptions = ActivityTrackingOptions.Baggage)
+                    .AddProvider(loggerProvider);
+            });
+
+            var logger = loggerFactory.CreateLogger("Logger");
+
+            Activity activity = new Activity("ScopeActivity");
+            activity.AddBaggage("testKey1", null);
+            activity.AddBaggage("testKey2", string.Empty);
+            activity.AddBaggage("testKey3", "testValue");
+            activity.Start();
+
+            logger.LogInformation("Message1");
+
+            activity.Stop();
+
+            foreach (string s in loggerProvider.LogText)
+            {
+                System.Console.WriteLine(s);
+            }
+
+            Assert.Equal("Message1", loggerProvider.LogText[0]);
+            Assert.Equal("testKey3:testValue, testKey2:, testKey1:", loggerProvider.LogText[2]);
+        }
+
+        [Fact]
+        public void TestActivityTrackingOptions_ShouldAddTagsAsNewScope_WhenTagsOptionIsSet()
+        {
+            var loggerProvider = new ExternalScopeLoggerProvider();
+
+            var loggerFactory = LoggerFactory.Create(builder =>
+            {
+                builder
+                    .Configure(o => o.ActivityTrackingOptions = ActivityTrackingOptions.TraceId | ActivityTrackingOptions.Tags)
+                    .AddProvider(loggerProvider);
+            });
+
+            var logger = loggerFactory.CreateLogger("Logger");
+
+            Activity activity = new Activity("ScopeActivity");
+            activity.AddTag("testKey1", null);
+            activity.AddTag("testKey2", string.Empty);
+            activity.AddTag("testKey3", "testValue");
+            activity.AddTag("testKey4", new Dummy());
+            activity.Start();
+
+            logger.LogInformation("Message1");
+
+            activity.Stop();
+
+            Assert.Equal("Message1", loggerProvider.LogText[0]);
+            Assert.Equal("testKey1:, testKey2:, testKey3:testValue, testKey4:DummyToString", loggerProvider.LogText[2]);
+        }
```

Comment:
This is how it looks like if the value is null or string.Empty.
Its how you suggested to do. However, I noticed that the Activity does something different for tags:
https://github.com/dotnet/runtime/blob/79ae74f5ca5c8a6fe3a48935e85bd7374959c570/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs#L1614

IMHO we should do the same here. WDYT?

Reply:
I am not sure I follow here. The line you are pointing at inside the Activity class is just filtering to the string tags and not caring really about the null values. What exactly you are suggesting here?

Reply:
Oh, sorry. I mixed things up.

However, the question remains if we should have the same behavior for logging as it is implemented here (i.e., only log tag values which are strings) or call `ToString()`.
Its currently implemented to call `ToString()` and this is what I would expect from the logger from my perspective as a consumer.
Why I'm asking then?

Because the summary comment of the Tags proprety states: https://github.com/dotnet/runtime/blob/79ae74f5ca5c8a6fe3a48935e85bd7374959c570/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs#L272-L274

But its not true. We enumerate TagObjects and do not log only string values.
But it is

Reply:
The comment is true for `Activity.Tags` property which expose only string-string key-value. `Activity.Tagsbjects` is string-object key-value. `Activity.Tags` is just legacy property which we kept because we cannot remove this API. `Activity.Tagsbjects` is the property that users should use all the time (which conform to the Open Telemetry specification).

> However, the question remains if we should have the same behavior for logging as it is implemented here (i.e., only log tag values which are strings) or call ToString().

Should call ToString() on the values. Logging string values only is wrong. Imagin, I add a tag like ('ErrorCode', 401) which has integer value. not logging this entry would be missing essential information in the logs.

Reply:
Okay, I see. Thank you for the clarification. This isn't related to the PR but are there plans to marke it as obsolete? As this API we are currently creating doesn't exist, yet we use the `Activity.Tags` in our `ConsoleFormatter` directly and thus we noticed exactly what you describe in your second comment. The int-values were not logged (which wasn't expected from a user perspective). As you say that the API shouldn't be used anymore and it's IMHO does something unexpected it would be helpful to get a warning that its obsolete.

Reply:
Yes, we can think obsoleting it or adding analyzer rule for it.

Reply:
I can create a ticket if you wish. Just let me know.